### PR TITLE
Add coverage support to C++ regression tests

### PR DESCRIPTION
**Pull request by fmeum**
Co-authored-by: Zhen Yu Ding <zhendeveloper@gmail.com>

Closes #174

### CODE REVIEW

**Comment by fmeum** [comment]
@stefanbucur Could you review this? This differs from #174 in that the LCOV merge is no longer build without `bazel coverage`.

@zhenyudg I added you as a coauthor.

**Comment by fmeum** [comment]
@stefanbucur Friendly ping

**Comment by stefanbucur** [comment]
Thanks so much for the ping, it totally fell through the cracks. I'll take a look by EOD.

(PS: Please feel free to ping me if you don't get an answer within a few days. Nowadays I'm spending less time on Github so things may easily slip off my attention.)

**Review by stefanbucur — CHANGES_REQUESTED** [review]
This is great - thanks so much for figuring out a way to do this cleanly. Just small-ish comments / suggestions.

File: fuzzing/tools/noop_lcov_merger.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env sh
```
[the remark this answers is not in the document]
Comment (fmeum) [reply fuzzing/tools/noop_lcov_merger.sh:1]: Added a comment.

File: fuzzing/tools/BUILD [hunk 2]
```diff
@@ -47,6 +47,25 @@ py_binary(
     ],
 )
 
+sh_binary(
+    name = "noop_lcov_merger",
+    srcs = ["noop_lcov_merger.sh"],
+)
+
+config_setting(
+    name = "collect_code_coverage",
```
[the remark this answers is not in the document]
Comment (fmeum) [reply fuzzing/tools/BUILD]: Done

**Review by stefanbucur — APPROVED** [review]
LGTM, but we seem to have new test failures?

**Comment by fmeum** [comment]
> LGTM, but we seem to have new test failures?

Looks like Bazel CI replaced `python` with `python3`, which is incompatible with Bazel 4. I will try raising the minimum version and see whether that fixes the failure.

**Comment by fmeum** [comment]
@stefanbucur I updated the Bazel version to 5.0.0, which makes Bazel look for `python3` in `PATH`. The [CI run still fails](https://github.com/bazelbuild/rules_fuzzing/actions/runs/3467075874/jobs/5791585426#step:4:239), but I when I run `/usr/bin/env python3` in `base-builder` locally, that passes. Not sure what's wrong here.

**Comment by stefanbucur** [comment]
> @stefanbucur I updated the Bazel version to 5.0.0, which makes Bazel look for `python3` in `PATH`. The [CI run still fails](https://github.com/bazelbuild/rules_fuzzing/actions/runs/3467075874/jobs/5791585426#step:4:239), but when I run `/usr/bin/env python3` in `base-builder` locally, that passes. Not sure what's wrong here.

Interesting indeed 🤔 FWIW, the CI fuzzing test has been broken for about 3 weeks at head too: https://github.com/bazelbuild/rules_fuzzing/actions/workflows/oss_fuzz.yml

**Comment by fmeum** [comment]
@stefanbucur Tests are passing now.

**Pull request merged**